sdk/schemaregistry/schema-registry-avro/review/schema-registry-avro.api.md

```diff
@@ -0,0 +1,27 @@
+## API Report File for "@azure/schema-registry-avro"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { SchemaRegistry } from '@azure/schema-registry';
+
+// @public
+export class SchemaRegistryAvroSerializer {
+    constructor(registry: SchemaRegistry, schemaGroup: string, options?: SchemaRegistryAvroSerializerOptions);
+    autoRegisterSchemas: boolean;
+    deserialize(buffer: Buffer): Promise<any>;
```

Comment:
Can we use generics so that the caller can hint what the return shape is?

```suggestion
    deserialize<T>(buffer: Buffer): Promise<T>;
```

Reply:
Interesting. I didn't realize you could call a method defined that way without a type argument and nothing in the parameters to infer it from. (C# bias.) I see you get unknown back in that case.

My concern with this change, which may be off due to my lack of familiarity with TypeScript idioms, is that it moves a cast into our code, and might give an illusion of safety that isn't there?

Reply:
You're right that it's pretty weak type safety, it's just sometimes easier to remember to write `deserialize<MyType>()`  which pins it down to something immediately. Usually at the point when you call deserialize you're hopefully thinking "what type should this be?"

Otherwise, if it returns `any` maybe you write `const foo = deserialize()` and now your variable is any (but you don't see that it's any, because it's inferred) and when you pass it around later in your function it always works, even in cases where it shouldn't.

I think the concern is that people don't normally type the result of calling a function, so `const foo:MyType = deserialize()` is less likely to get written, which is the only other way you'd catch such an error.

Reply:
That makes sense. I'm convinced and I'll change it. 😃